File: examples/new_test.py
from prometeo.linalg import *
import sys 

n: int = 10

A: prmt_mat = prmt_mat(n, n)
for i in range(n):
    for j in range(n):
        A[i][j] = i*n + j 

B: prmt_mat = prmt_mat(n, n)
for i in range(n):
    for j in range(n):
        B[i][j] = 0.0 

for i in range(n):
    B[i][i] = 1.0 

C: prmt_mat = prmt_mat(n, n)
for i in range(n):
    for j in range(n):
        C[i][j] = 0.0 

i: int = 0.0
while i < n:
    i = i + 1

if i >= 5:
    i = i *2

prmt_gemm_nt(A, B, C, C)

# print results
print('\n\nB = ')
prmt_print(B)
print('\n\nA = ')
prmt_print(A)
print('\n\nC = ')
prmt_print(C)

File: examples/new_test.c
#include "new_test.h"
int n = 10;
struct prmt_mat * A = prmt_mat(n, n);
for (int i = 0; i < n; i++) {
    for (int j = 0; j < n; j++) {
        prmt_mat_set_el(A, i, j, (i * n + j));
}
}
struct prmt_mat * B = prmt_mat(n, n);
for (int i = 0; i < n; i++) {
    for (int j = 0; j < n; j++) {
        prmt_mat_set_el(B, i, j, (0.0));
}
}
for (int i = 0; i < n; i++) {
    prmt_mat_set_el(B, i, i, (1.0));
}
struct prmt_mat * C = prmt_mat(n, n);
for (int i = 0; i < n; i++) {
    for (int j = 0; j < n; j++) {
        prmt_mat_set_el(C, i, j, (0.0));
}
}
int i = 0.0;
while(i < n) {
    i = i + 1;
}
if(i >= 5) {
    i = i * 2;
}
prmt_gemm_nt(A, B, C, C);
print('\n\nB = ');
prmt_print(B);
print('\n\nA = ');
prmt_print(A);
print('\n\nC = ');
prmt_print(C);
